lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js

```diff
@@ -228,7 +228,7 @@ export default function BasicLTILaunchApp() {
       // If a teacher launches an assignment or the LMS does not support reporting
       // outcomes or grading is not enabled for the assignment, then no submission
       // URL will be available.
-      if (!canvas.speedGrader || !canvas.speedGrader.submissionParams) {
+      if (!canvas.speedGrader) {
```

Comment:
The backend code either doesn't set `speedGrader`, or it sets a `speedGrader` object with `submissionParams`. See https://github.com/hypothesis/lms/blob/0edc7c52da3aecc939fb7d433cb92ca44c3bc474/lms/resources/_js_config/__init__.py#L347.

Reply:
This conditional rings a faint bell. Are you absolutely certain about the backend behavior in all cases? Ping @marcospri

Reply:
This is the only place that `submissionParams` gets set: https://github.com/hypothesis/lms/blob/8086c9b7e979d8486331a2873bd894c076e5cf16/lms/resources/_js_config/__init__.py#L346.

If that changes in future, we'll update the `ConfigObject` type in the frontend and then update other code as needed.

Reply:
But is it the only place that `speedGrader` gets set? If so, OK!

Reply:
That looks right, this one is conditional below:

https://github.com/hypothesis/lms/blob/8086c9b7e979d8486331a2873bd894c076e5cf16/lms/resources/_js_config/__init__.py#L367